clang/test/dpct/helper_files_ref/include/dnnl_utils.hpp

```diff
@@ -408,158 +724,680 @@ class engine_ext {
-/// \param [out] dst Pointer to destination data.
-  void sum(float alpha, const memory_desc_ext &src_desc, void *src, float beta,
-           const memory_desc_ext &dst_desc, void *dst);
-/// Computing a specified activation function value.
-/// \param [in] desc Activation descriptor.
-/// \param [in] alpha Pointer to scaling factors used to scale the computed value.
-/// \param [in] src_desc Source memory descriptor.
-/// \param [in] src Pointer to source data.
-/// \param [in] beta Pointer to scaling factors used to scale the prior value in the destination memory.
-/// \param [in] dst_desc Destination memory descriptor.
-/// \param [out] dst Pointer to destination data.
-  void activation_forward(activation_desc &desc, float alpha,
-                          const memory_desc_ext &src_desc, void *src, float beta,
-                          const memory_desc_ext &dst_desc, void *dst);
-/// Computing the gradient of a specified activation function.
-/// \param [in] desc Activation descriptor.
-/// \param [in] alpha Pointer to scaling factors used to scale the computed value.
-/// \param [in] dst_desc Destination memory descriptor.
-/// \param [in] dst Pointer to destination data.
-/// \param [in] diff_dst_desc Differential destination memory descriptor.
-/// \param [in] diff_dst Pointer to differential destination data.
-/// \param [in] src_desc Source memory descriptor.
-/// \param [in] src Pointer to source data.
-/// \param [in] beta Pointer to scaling factors used to scale the prior value in the differential destination memory.
-/// \param [in] diff_src_desc Differential source memory descriptor.
-/// \param [out] diff_src Pointer to differential source data.
-  void activation_backward(activation_desc &desc, float alpha,
+  /// Retrieving the user's SYCL queue set in the oneDNN engine.
+  /// \returns Pointer to the SYCL queue.
+  sycl::queue *get_queue() const { return _q; }
+  /// Setting all elements of a memory to a given value.
+  /// \param [in] src_desc Source memory descriptor.
+  /// \param [in] src Pointer to source data.
+  /// \param [in] valuePtr Pointer to a signle value.
+  /// \returns An event representing the fill operations.
+  sycl::event fill(const memory_desc_ext &src_desc, void *src,
+                   const void *valuePtr);
+  /// Coping the scaled data from a memory to another memory with a different
+  /// description.
+  /// \param [in] alpha Value to scaling factors used to scale the computed
+  /// value.
+  /// \param [in] src_desc Source memory descriptor.
+  /// \param [in] src Pointer to source data.
+  /// \param [in] beta Value to scaling factors used to scale the prior value
+  /// in the destination memory.
+  /// \param [in] dst_desc Destination memory descriptor.
+  /// \param [out] dst Pointer to destination data.
+  /// \returns An event representing the reorder operations.
+  sycl::event reorder(float alpha, const memory_desc_ext &src_desc, void *src,
```

Comment:
the change break the backward compatibility,  let's keep both async and sync interface.